### ISSUE TITLE
setup.cfg ignored

### ISSUE THREAD

**Issue by oerp-odoo** [SublimeLinter/SublimeLinter-pydocstyle#24]
I have this config called `setup.cfg`:

```
[pydocstyle]
ignore = D100
```
I have directory `source` where are my projects located and `setup.cfg` file too. Like this:
```
source/
  project1
  project2
  setup.cfg
```
When I open sublime and it runs linter, I see that it ignores my `setup.cfg` and still throws warning about `D100`:

```
blimeLinter: #2 linter.py:1263: Running ...

  /home/oerp/odoo11/source/personal/oerp-odoo  (working dir)
  $ /usr/local/bin/pydocstyle /tmp/tmpi74t6k.py

[2018-08-11 16:43:11,500 - INFO     - SublimeLinter.lint.linter] Running ...

  /home/oerp/odoo11/source/personal/oerp-odoo  (working dir)
  $ /usr/local/bin/pydocstyle /tmp/tmpi74t6k.py

SublimeLinter: #2 linter.py:955: pydocstyle output:
    /tmp/tmpi74t6k.py:1 at module level:
            D100: Missing docstring in public module
[2018-08-11 16:43:11,712 - INFO     - SublimeLinter.lint.linter] pydocstyle output:
    /tmp/tmpi74t6k.py:1 at module level:
            D100: Missing docstring in public module
```
Though if I manually run linter like : `pydocstyle machine/models/machine.py`. It does check my `setup.cfg` and does not throw warning about D100 (for testing purposes I ran it from same directory as sublime linter indicated looking -> `/home/oerp/odoo11/source/personal/oerp-odoo`).

P.S. Though for example I have `eslint` configuration which uses same approach and it works. I wonder why this one is ignored?

**Comment by kaste** [comment]
Hi!

`pydocstyle` uses the file to search for a config upwards. Since we're using tempfiles here, this cannot succeed. The workaround would be to set `config` to the actual config file you want to use. 

We could change this linter to work with the real files on disk only. (It would only run on save, not in the background anymore.)

There is also a dead PR over there which will enable support via stdin. In that case, the proposal is that the user has to specify the config manually bc `pydocstyle` will not search for it at all.

**Comment by oerp-odoo** [comment]
Hi,
hmm, I did not get where then I need to specify configuration? I read that it is no longer supported to specify configuration (can't find a link now..) directly in sublime and it was recommended to use config file like the one I tried. Or you are talking about some other configuration?

**Comment by kaste** [comment]
For project settings, you would set 

```
	"settings": {
		"SublimeLinter.linters.pydocstyle.config": "path/to/setup.cfg"
	}
```

Or globally, in the SublimeLinter user settings:

```
	linters: {
		"pydocstyle": {
			"config": "$folder/setup.cfg"
		}
	}
```
